Jenkinsfile

```diff
@@ -169,41 +192,38 @@ pipeline {
                             deleteDir()
                             unstash 'CIScripts'
                             powershell script: './CIScripts/Test.ps1'
+
+                            // TODO: you can do better
+                            stash name: 'testReport', includes: 'CIScripts/Test/testReport.xml'
                         }
                     }
                 }
             }
-        }
-    }
 
-    environment {
-        LOG_SERVER = "logs.opencontrail.org"
-        LOG_SERVER_USER = "zuul-win"
-        LOG_ROOT_DIR = "/var/www/logs/winci"
-        BUILD_SPECIFIC_DIR = "${ZUUL_UUID}"
-        JOB_SUBPATH = env.JOB_NAME.replaceAll("/", "/job/")
-        REMOTE_DST_FILE = "${LOG_ROOT_DIR}/${BUILD_SPECIFIC_DIR}/log.txt.gz"
+            post {
+                always {
+                    node('master') {
+                        script {
+                            try {
+                                unstash 'testReport'
+                                sh "rsync CIScripts/Test/testReport.xml ${logServerUser}@${logServer}:${logsRemoteDir}"
+                            } catch (err) {
+                                echo "No test reports available"
+                            }
+                        }
+                    }
+                }
+            }
+        }
     }
 
     post {
         always {
             node('master') {
                 script {
-                    // Job triggered by Zuul -> upload log file to public server.
-                    // Job triggered by Github CI repository (variable "ghprbPullId" exists) -> keep log "private".
-
-                    // TODO JUNIPER_WINDOWSSTUBS variable check is temporary and should be removed once
-                    // repository contrail-windows is accessible from Gerrit and it is main source of
-                    // windowsstubs code.
-                    if (env.ghprbPullId == null && env.JUNIPER_WINDOWSSTUBS == null) {
-                        // unstash "buildLogs"
-                        // TODO correct flags for rsync
-                        sh "ssh ${LOG_SERVER_USER}@${LOG_SERVER} \"mkdir -p ${LOG_ROOT_DIR}/${BUILD_SPECIFIC_DIR}\""
-                        def tmpLogFilename = "clean_log.txt.gz"
-                        obtainLogFile(env.JOB_NAME, env.BUILD_ID, tmpLogFilename)
-                        sh "rsync $tmpLogFilename ${LOG_SERVER_USER}@${LOG_SERVER}:${REMOTE_DST_FILE}"
-                        deleteDir()
-                    }
+                    def tmpLogFilename = "clean_log.txt.gz"
+                    obtainLogFile(env.JOB_NAME, env.BUILD_ID, tmpLogFilename)
+                    sh "rsync $tmpLogFilename ${logServerUser}@${logServer}:${logsDestinationFile}"
```

Comment:
IMHO this whole `sh "rsync $tmpLogFilename ${logServerUser}@${logServer}:` should be abstracted into a function`pushToLogServer(filename, relative_filepath_on_the_fileserver)`

Then we can do magic like:

```
pushToLogServer(tmpLogFilename, "/log.txt.gz")
pushToLogServer("CIScripts/Test/testReport.xml", "testReports/testReport.xml")
```